### PR TITLE
Do not use Cockpit manifest for storing supported languages

### PR DESCRIPTION
**Pull request by lslezak**
## Problem

- Continue with dropping the Cockpit dependency

## Solution

- Do not use the Cockpit manifest for storing the list of supported languages
- Use a separate `languages.json` file instead of the `manifest.json`
- Import the JSON file directly into the JS code
- The workflow is not changed, the update script just generates a different file
- The change is backward compatible, works with both Cockpit and the new server

## Testing

- Updated unit tests
- Tested manually, the language selection still works fine

### CODE REVIEW

File: web/share/update-languages.py [hunk 1]
```diff
@@ -124,32 +118,31 @@ def update(self, po_files, lang2territory: str, threshold: int):
...
             include_territory = languages.count(locale.language) > 1
-            self.content["locales"][locale.code()] = locale.name(
-                include_territory)
+            self.content[locale.code()] = locale.name(include_territory)
 
     def write(self):
         with open(self.path, "w+") as file:
-            json.dump(self.content, file, indent=4, ensure_ascii=False)
+            json.dump(self.content, file, indent=4, ensure_ascii=False,
+                      sort_keys=True)
```
Comment (lslezak) [inline web/share/update-languages.py:128]: The `sort_keys=True` option ensures the list is sorted and avoids random reordering of the items.

File: web/tsconfig.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
     "noEmit": true,
     "target": "esnext",
     "moduleResolution": "node",
+    "resolveJsonModule": true,
```
Comment (lslezak) [inline web/tsconfig.json:8]: To support loading JSON files.

File: web/webpack.config.js [hunk 3]
```diff
@@ -112,16 +111,6 @@ module.exports = {
         // ignore SSL problems (self-signed certificate)
         secure: false,
       },
-      // forward the manifests.js request and patch the response with the
-      // current Agama manifest from the ./src/manifest.json file
-      "/manifests.js": {
```
Comment (lslezak) [inline web/webpack.config.js:117]: We do not need to read  the manifests from the server anymore.

File: web/share/update-languages.py [hunk 4]
```diff
@@ -124,32 +118,31 @@ def update(self, po_files, lang2territory: str, threshold: int):
...
 
 
 if __name__ == "__main__":
-    parser = ArgumentParser(prog="locales.py")
-    parser.set_defaults(func=update_manifest)
+    parser = ArgumentParser(prog="update-languages.py")
+    parser.set_defaults(func=update_languages)
     parser.add_argument(
-        "manifest", type=str, help="Path to the manifest file",
+        "file", type=str, help="Path to the output file",
```
Comment (jreidinger) [inline web/share/update-languages.py:145]: I would call it output instead of file as it is more similar to input file.

File: web/cspell.json [hunk 5]
```diff
@@ -3,6 +3,7 @@
   "language": "en",
   "allowCompoundWords": false,
   "ignorePaths": [
+    "src/languages.json",
```
Comment (jreidinger) [inline web/cspell.json:6]: maybe add it also to `.gitignore`?

File: .github/workflows/weblate-merge-po.yml [hunk 6]
```diff
@@ -92,7 +92,7 @@ jobs:
         if: steps.check_changes.outputs.po_updated == 'true'
         working-directory: ./agama
         run: |
-          web/share/update-manifest.py web/src/manifest.json
+          web/share/update-languages.py web/src/languages.json
```
Comment (jreidinger) [inline .github/workflows/weblate-merge-po.yml:95]: why not generate it to stdout by default and allow redirect?

File: web/share/update-languages.py (hunk 4 above)
Comment (jreidinger) [reply web/share/update-languages.py:145]: or maybe even better write to stdout by default and let user redirect it where he needs?

**Review by dgdavid — COMMENTED** [review]
In general it looks good. I guess it deserves an entry in the changes file too.

**Review by dgdavid — APPROVED** [review]
:+1:

File: web/share/update-languages.py [hunk 7]
```diff
@@ -105,14 +109,16 @@ def update(self, po_files, lang2territory: str, threshold: int):
...
+                    file=sys.stderr
                 )
             elif po_file.coverage() < threshold:
                 print(
                     "not enough coverage for '{language}' ({coverage}%)"
                     .format(
                         language=locale.code(),
-                        coverage=po_file.coverage())
+                        coverage=po_file.coverage()),
+                    file=sys.stderr
```
Comment (jreidinger) [inline web/share/update-languages.py:121]: :+1: for writting errors to stderr

**Pull request merged**